base/system/services/database.c

```diff
@@ -1311,199 +1311,31 @@ ScmGetBootAndSystemDriverState(VOID)
 
 
 DWORD
-ScmControlService(HANDLE hControlPipe,
```

Comment:
Is this PR meant to be a full replacement of Dougs parallel PR7375? 100% Superseding it? And would fix the Service-stopping also on its own? In this case I think it is legit to open it. But **if** in contrast it would not fix the stuff that Doug intends to fix with his PR, then I would consider it to be highly unfriendly to open this refactoring PR now, creating potential collisions with Dougs work! I would then tend to reject this PR at the current point in time. Which one of both is it?

Reply:
No it's not a replacement of @DougLyons PR. But it just happens that he also modified some parameters of ScmControlService(), which wasn't necessary in principle to fix services stopping.

Reply:
I do understand and kind of agree that the params-list-change actually was not necessary in Dougs PR. Nevertheless please wait with merging your PR until Dougs PR will have been merged. Resolving conflicts should lay on your shoulders in this specific case, not on his.

Reply:
For the record I posted two very small fixes (2-4 lines or changes) to the delay in services stopping in https://jira.reactos.org/browse/CORE-16949. But PR #1225 had a concept of reference counting that seemed good to me, so I went back to it to try and work from that point. I agree with the principle of this PR though and it reduces lines-of-code, so that is good. I did look at the changes to the parameters, but it seemed best to me to take the complete SERVICE structure along in case other parts might be needed later. Thanks.

Reply:
@Doug-Lyons personally I don't have a strong preference whether _just_ "the pipes handle + the services name" is passed there, or whether the full PSERVICE is passed there, which you do consider to be more flexible. I am kind of neutral in that regard. But seriously: Passing both: PSERVICE and then the pServiceName as well (which is the current state of your PR7375) does indeed make no sense at all. Either pass "the pipes handle + the services name" or do pass "PSERVICE with**out** the pServiceName" then.

Reply:
> but it seemed best to me to take the complete SERVICE structure along in case other parts might be needed later

No, the other SERVICE struct members won't be necessary later. The purpose of ScmControlService is just to take few stuff and package them to be sent to the pipe to the service (via the service event loop inside advapi32).

Reply:
I have now changed ScmControlService's first parameter to 'HANDLE hControlPipe' based on reviewer comments for #7375. Thanks.